SignalServiceKit/src/Util/OWSAnalytics.m

```diff
@@ -145,46 +162,103 @@ - (void)tryToSyncEvents
             if (!firstEventKey) {
                 return;
             }
-
+            
             firstEventDictionary = [transaction objectForKey:firstEventKey inCollection:kOWSAnalytics_EventsCollection];
             OWSAssert(firstEventDictionary);
             OWSAssert([firstEventDictionary isKindOfClass:[NSDictionary class]]);
         }];
 
-        if (!firstEventDictionary) {
-            return;
+        if (firstEventDictionary) {
+            [self sendEvent:firstEventDictionary eventKey:firstEventKey isCritical:NO];
         }
+    });
+}
 
-        DDLogDebug(@"%@ trying to deliver event: %@", self.tag, firstEventKey);
+- (void)sendEvent:(NSDictionary *)eventDictionary eventKey:(NSString *)eventKey isCritical:(BOOL)isCritical
+{
+    OWSAssert(eventDictionary);
+    OWSAssert(eventKey);
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    OWSAssert(dispatch_get_current_queue() == self.serialQueue);
+#pragma clang diagnostic pop
+
+    if (isCritical) {
+        [self submitEvent:eventDictionary
+            eventKey:eventKey
+            success:^{
+                DDLogDebug(@"%@ sendEvent[critical] succeeded: %@", self.tag, eventKey);
```

Comment:
Though I think in all the current cases we'll be crashing shortly after this point, but that's not necessarily intuitive. Should we reset hasRequestInFlight here as well in case there's a situation where multiple critical events are being fired?